src/components/TimeSelectInput/index.js

```diff
@@ -0,0 +1,503 @@
+        }
+        if (inputFocusedIndex === 2) {
+            this.setNextAmPmValue();
+        }
+    }
+
+    handleButtonsDown() {
+        this.isUpOrDownButtonPressed = true;
+    }
+
+    handleButtonsFocus() {
+        const { inputFocusedIndex } = this.state;
+        const currentInputFocused = this.inputsMap[inputFocusedIndex];
+        if (currentInputFocused) {
+            currentInputFocused.current.focus();
+        }
+    }
+
+    resetState() {
+        const { inputFocusedIndex } = this.state;
+        if (inputFocusedIndex === 0) {
+            this.handleChangeTime()({
+                hour: '',
+            });
+            this.prevHour = '';
+        }
+        if (inputFocusedIndex === 1) {
+            this.handleChangeTime({
+                minutes: '',
+            });
+            this.prevMinutes = '';
+        }
+    }
+
+    updateTime() {
+        const { value } = this.props;
+        this.setState({
+            hour: getHour(value),
+            minutes: getMinutes(value),
+            ampm: getAmPm(value),
+        });
+    }
+
+    focusHourInput() {
+        this.hourInputRef.current.focus();
+        this.setState({ inputFocusedIndex: 0 });
+    }
+
+    incrementHour() {
+        const { hour } = this.state;
+        const { hour24 } = this.props;
+        const hourValue = hour || this.prevHour;
+        this.handleChangeTime({
+            hour: normalizeHour(getNextHour(hourValue, hour24), hour24),
+        });
+    }
+
+    decrementHour() {
+        const { hour } = this.state;
+        const { hour24 } = this.props;
+        const hourValue = hour || this.prevHour;
+        this.handleChangeTime({
+            hour: normalizeHour(getPrevHour(hourValue, hour24), hour24),
+        });
+    }
+
+    incrementMinutes() {
+        const { minutes } = this.state;
+        const minutesValue = minutes || this.prevMinutes;
+        this.handleChangeTime({
+            minutes: normalizeMinutes(getNextMinute(minutesValue)),
+        });
+    }
+
+    decrementMinutes() {
+        const { minutes } = this.state;
+        const minutesValue = minutes || this.prevMinutes;
+        this.handleChangeTime({
+            minutes: normalizeMinutes(getPrevMinute(minutesValue)),
+        });
+    }
+
+    handleChangeTime(newState) {
+        const { hour, minutes, ampm } = this.state;
+        const { onChange, hour24 } = this.props;
+        const currentHour = newState.hour || hour || this.prevHour;
+        const currentMinutes = newState.minutes || minutes || this.prevMinutes;
+        const currentAmPm = newState.ampm || ampm;
+        const time = get24HourTime({
+            hour: currentHour,
+            minutes: currentMinutes,
+            ampm: currentAmPm,
+        });
+        this.setState(newState);
+
+        if (
+            (currentHour && currentMinutes && ampm && !hour24) ||
+            (currentHour && currentMinutes && hour24)
+        ) {
+            onChange(time);
```

Comment:
@rafaelmg83  The behavior of all our Input components is, that when you change the value you have to call the onChange method. You have to check that behavior well in this component, it is working fine.
I already told you to change the examples to reflect that behavior, so you can try adding a console.log (value) in the onChange and check that every time you change the date in the component it makes you console.log.

Reply:
I don't know what you mean exactly, but the onChange event is called every time that any part of hour, minutes or am/pm changes except the value is incomplete where it has no logic to call it. Of course I tried the console.log test before push. Maybe you are seeing something that I am not, but I think that behavior is fine as it is.